frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -84,9 +84,9 @@ export const getSurveyDataQuery = (surveyName: string): DataTableNode => {
     return surveyDataQuery
 }
 
-export const getSurveyMetricsQueries = (surveyId: string, surveyName: string): SurveyMetricsQueries => {
-    const surveysShownHogqlQuery = `select count() as 'survey shown' from events where event == '${surveyName} survey shown' and properties.$survey_id == '${surveyId}'`
-    const surveysDismissedHogqlQuery = `select count() as 'survey dismissed' from events where event == '${surveyName} survey dismissed' and properties.$survey_id == '${surveyId}'`
+export const getSurveyMetricsQueries = (surveyId: string): SurveyMetricsQueries => {
+    const surveysShownHogqlQuery = `select count() as 'survey shown' from events where event like '%shown%' and properties.$survey_id == '${surveyId}'`
```

Comment:
This works, but can be much slower compared to what we had before, because  can't effectively use the sort key. Also since it's querying all time(?) - actually, is it querying all time? I'm not sure if hogql adds barriers to queries automatically.

Which is why, what I meant instead was: make everything & everyone send the event called 'survey shown', 'survey dismissed' & 'survey response' and filter on `survey_id` like you are for results / counts.

Reply:
Saw your comment here: https://github.com/PostHog/posthog/pull/16165#discussion_r1237049544

Makes some sense! I'm not sure if users will ever have to enter this event at all, since we're handling all of this for them + can get pretty confusing when name changes but events still have old survey name.

But in any case, if you want to go for this, it's fine, keep this, the perf hit here shouldn't be too big.

Just making the matching more precise:
```suggestion
    const surveysShownHogqlQuery = `select count() as 'survey shown' from events where event like '%survey shown%' and properties.$survey_id == '${surveyId}'`
```

Reply:
updated :)